lib/fastlane/plugin/wpmreleasetoolkit/helper/git_helper.rb

```diff
@@ -179,6 +179,21 @@ def self.fetch_all_tags
         Action.sh('git', 'fetch', '--tags')
       end
 
+      # Counts the number of commits between two references in the Git repository.
+      #
+      # @param base_ref [String] The base reference (branch name, tag, or commit hash).
+      # @param head_ref [String] The head reference (branch name, tag, or commit hash).
+      #
+      # @return [Integer] The number of commits between base_ref and head_ref.
+      #
+      def self.count_commits_between(base_ref:, head_ref:)
+        git_repo = Git.open(Dir.pwd)
+
+        new_commits = git_repo.log.between(base_ref, head_ref)
```

Comment:
💭  This might return the wrong number of commits if there are merge commits (aka commits with more than one parent) along the path between the two points.

For example, how many commits are there between `A` and `G` here?

```
A---B---C---M---F---G
     \     /
      D---E
```

 - 5 (B, C, M, F, G)?
 - 6 (B, D, E, M, F, G)?
 - 7 (B, C, D, E, M, F, G — counting the total on both paths)?

This is why I added `--first-parent` in the command I suggested to use in the apps-infra-roadmap issue 😉 

You will want to add a unit test for that case too btw.

Reply:
I've changed the implementation a bit, making the `--first-parent`-like implementation optional, and adding this case to the Spec setup: 9210a33f910d46d77bb2ffcc883f165b73e1cbe8.

Reply:
Sorry for the back and forth, but the more I think about it the more I wonder if there's a use case for counting the number of commits between two SHA… that is anything else than then checking if that count is >0 or not. Be it to decide if we need to do a new beta or not, or decide if a PR is going to be empty because would be asking GitHub to merge a commit into itself… in both cases testing if the two commit resolve to the same sha1 would be enough… so I might have suggested you a direction that was more complicated that it needed to, sorry.

Like, maybe testing if `git rev-parse #{base}` and `git rev-parse `#{head}` (or `head_commit.sha == base_commit.sha1`, in the Git gem parlance) is all we need after all 😓

Reply:
I've made some changes (4d19508443dbeb7303f289d2efe95dd2e4bd692a) as discussed via DM:
- Updated the ref check to be a boolean (`count_commits_between` to `has_commits_between?`)
- Check whether there's already an intermediate branch
- Move the check `has_commits_between?` to after the callback, to cover cases where the callback added new commits